orchagent/bufferorch.cpp

```diff
@@ -974,7 +988,52 @@ task_process_status BufferOrch::processQueue(KeyOpFieldsValuesTuple &tuple)
             if (need_update_sai)
             {
                 SWSS_LOG_DEBUG("Applying buffer profile:0x%" PRIx64 " to queue index:%zd, queue sai_id:0x%" PRIx64, sai_buffer_profile, ind, queue_id);
-                sai_status_t sai_status = sai_queue_api->set_queue_attribute(queue_id, &attr);
+            }
+
+            QueueTask::QueueContext queueContext;
+            queueContext.queue_id = queue_id;
+            queueContext.attr = SaiAttrWrapper(SAI_OBJECT_TYPE_QUEUE, attr);
+            queueContext.counter_was_added = counter_was_added;
+            queueContext.counter_needs_to_add = counter_needs_to_add;
+            queueContext.index = ind;
+            queueContext.update_sai = need_update_sai;
+
+            portContext.queues.emplace_back(queueContext);
+        }
+
+        task.ports.emplace_back(portContext);
+    }
+
+    m_queueBulk[op].emplace_back(task);
+
+    return task_process_status::task_success;
+}
+
+task_process_status BufferOrch::processQueuePost(const QueueTask& task)
+{
+    SWSS_LOG_ENTER();
+
+    const auto& key = kfvKey(task.kofvs);
+    const auto& op = kfvOp(task.kofvs);
+    const auto tokens = tokenize(key, delimiter);
+    Port port;
+
+    for (const auto& portContext: task.ports)
```

Comment:
Any capability check to query vendor SAI's bulk API implementation?

Reply:
Not required, please check https://github.com/jianyuewu/sonic-sairedis/blob/master/syncd/Syncd.cpp#L2186